src/org/jetbrains/kotlin/test/helper/actions/GeneratedTestComboBoxAction.kt

```diff
@@ -128,9 +129,18 @@ class GeneratedTestComboBoxAction(val baseEditor: TextEditor) : ComboBoxAction()
                 .submit(AppExecutorUtil.getAppExecutorService())
         }
 
+        private val VirtualFile.nameWithoutAllExtensions get() = name.split('.').first()
+
+        private val String.asPathWithoutAllExtensions: String
+            get() {
+                val pathParts = split(File.separator).toMutableList()
+                pathParts[pathParts.lastIndex] = pathParts[pathParts.lastIndex].split('.').first()
+                return pathParts.joinToString(File.separator)
```

Comment:
This looks a bit more optimal without unnecessary allocations:

```suggestion
        val lastFileSeparatorIndex = lastIndexOf(File.separator).let { if (it == -1) 0 else it }
        val extensionDotIndex = indexOf('.', lastFileSeparatorIndex).let { if (it == -1) length else it }
        return substring(0, extensionDotIndex)
```

Reply:
And possibly if we were to do `indexOfLast()` from back to front, then we could also avoid the second `.let{}`, because the result would be subtractable from `lastIndex`: in case of -1 it's `lastIndex - (-1) == length`, and then it just decreases down to give us the proper index. Thought that may be a bit unreadable already :)

Reply:
Ok, it's reasonable.

Reply:
Or maybe not. WDYT? The example below does still perform an allocation, but we can always implement a manual `reversedIndexOf()` if we want

```
val String.asPathWithoutAllExtensions: String
	get() {
    	val indexOfSeparator = indexOf(File.separator)//.also(::println)

        val indexOfFirstExtensionDot = reversed()
            .substring(0, lastIndex - indexOfSeparator)
            .lastIndexOf('.')
            //.also(::println)

        return substring(0, lastIndex - indexOfFirstExtensionDot)
	}
```

Reply:
Your last example works incorrectly because it doesn't consider dots in a folder name. For instance, something like this: `F:\tests.2\dontReportWarningAboutChangingExecutionOrderForVararg.fir.kt`.

Also, it looks too compilcated to me (call `reverse` looks confused for such a simple task).

Maybe it makes sense to write some tests on this extension to get rid of potential mistakes.

Reply:
> Your last example works incorrectly because it doesn't consider dots in a folder name. For instance, something like this: `F:\tests.2\dontReportWarningAboutChangingExecutionOrderForVararg.fir.kt`.

True, my typo. There should have been `val indexOfSeparator = lastIndexOf(File.separator)`, not `indexOf(File.separator)`

Reply:
I understand why this code may seem unintuitive, but you can think of it this way: in order to avoid having `-1`-related corner cases we must make sure that higher values of `indexOf()`-like searching functions correspond directly to shifting indices in one direction.

- The first thing we do is we search for the last file separator, and the more separators we have, the bigger should be the resulting index.
- The second thing we do is we search for the first `'.'` after the last separator which is the same as looking for the last `'.'` before the first separator, and the more extensions we have, the further we should move back from the end of the string.

So, if we for some reason do some `indexOf()` such that its higher values correspond to higher movement along some direction, but `-1` corresponds to the maximum movement along it, then it's a sign that we can look from the other side to possibly get rid of some corner-cased `ifs`.

But I do realize that this intuition may be uncommon

Reply:
The argument about string allocations is valid, but, again, this is not performance-critical code, and even if it were, there is a straghtforward way to implement `reversedIndexOf(char, startIndex)` which would do exactly the same as `indexOf()`, but return the index from the end rather than from the start.

Reply:
A pity that we don't have such a function in our stdlib, though

Reply:
Ok, let's wait for a final resolution from @demiurg906 or @cypressious 😊

Reply:
I have just realized that `reversedIndexOf()` itself can be implemented via `lastIndex - indexOfOrLength()` and the latter function reads way easier (in fact, it's you have inlined in your suggestion). Left it as a separate function, because at this point I believe it deserves it. See [c9c6ef2](https://github.com/demiurg906/test-data-helper-plugin/pull/34/commits/c9c6ef2145f539bd394df60c7bf3db07a78c85b1)